nixos/modules/services/web-apps/mealie.nix

```diff
@@ -59,21 +111,31 @@ in
 
       environment = {
         PRODUCTION = "true";
-        ALEMBIC_CONFIG_FILE="${pkg}/config/alembic.ini";
+        ALEMBIC_CONFIG_FILE = "${pkg}/config/alembic.ini";
         API_PORT = toString cfg.port;
         DATA_DIR = "/var/lib/mealie";
         CRF_MODEL_PATH = "/var/lib/mealie/model.crfmodel";
       } // (builtins.mapAttrs (_: val: toString val) cfg.settings);
 
       serviceConfig = {
         DynamicUser = true;
-        User = "mealie";
+        User = cfg.user;
         ExecStartPre = "${pkg}/libexec/init_db";
         ExecStart = "${lib.getExe pkg} -b ${cfg.listenAddress}:${builtins.toString cfg.port}";
         EnvironmentFile = lib.mkIf (cfg.credentialsFile != null) cfg.credentialsFile;
         StateDirectory = "mealie";
         StandardOutput="journal";
-      };
+      }
+        # If a custom data directory has been set, then allow the user mealie is
+        # running as to write to it. This is only necessary when systemd has
+        # dynamically generated a user to run mealie as. Otherwise, it is a no-op.
+        #
+        # Also, unset the state directory, as mealie won't be writing to it when
+        # a custom data directory is in use.
+        // lib.mkIf (environment.DATA_DIR != "/var/lib/mealie") {
```

Comment:
Nitpick: I would maybe put the `/var/lib/mealie` inside a variable somewhere so that if its value gets changed, we don't forget this condition